src/Stratis.Bitcoin.Features.Wallet/Wallet.cs

```diff
@@ -728,12 +728,29 @@ public IEnumerable<UnspentOutputReference> GetSpendableTransactions(int currentC
 
                     if (confirmationCount >= confirmations)
                     {
-                        yield return new UnspentOutputReference
+                        if (includeImmature || (!(transactionData.IsCoinBase ?? false) && !(transactionData.IsCoinStake ?? false)))
                         {
-                            Account = this,
-                            Address = address,
-                            Transaction = transactionData
-                        };
+                            // This output can unconditionally be included in the results.
+
+                            yield return new UnspentOutputReference
+                            {
+                                Account = this,
+                                Address = address,
+                                Transaction = transactionData
+                            };
+                        }
+
+                        if (!includeImmature && ((transactionData.IsCoinBase ?? false) || (transactionData.IsCoinStake ?? false)) && (confirmationCount >= consensus.CoinbaseMaturity))
```

Comment:
In any case it would be nice to separate the parts of the condition to variables and you could reuse that here and about